app/views/banners/_form.html.erb

```diff
@@ -26,6 +26,10 @@
       <%= form.label :active, "Active?", class: 'form-check-label' %>
       <span data-reveal-target="item" class="text-danger <%= conditionally_add_hidden_class(form.object.active) %>">Warning: This will replace your current active banner</span>
     </div>
+    <span class="input-style-1">
+      <%= form.label :expires_at, "Expires at (optional)" %>
+      <%= form.datetime_field :expires_at, value: banner.expires_at&.in_time_zone(cookies[:browser_time_zone]), required: false %>
```

Comment:
This logic should be in a view helper (and tested).
```
banner.expires_at&.in_time_zone(cookies[:browser_time_zone])
```
You might take a look at [datetime_local_field](https://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html#method-i-datetime_local_field). It might eliminate the need for the logic.

Reply:
Unfortunately I don't think datetime_local_field helps us here.  According to your link, it wraps [ActionView::Helpers::FormHelper#datetime_local_field](https://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-datetime_local_field) which in turn states that it's only an alias for `datetime_field`. According to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#setting_timezones, timezone aware logic was intentionally left out of this element.

That said, I did try to move this logic to `BannerHelper` but ultimately put it in the model. It kind of feels like it would fit best in a presenter type object but might be overkill for now? Any thoughts?

Reply:
Either model or presenter is fine. I agree presenter makes more sense, but if it's just one method does feel like overkill.